services/gps/trends.py

```
@@ -0,0 +1,435 @@
+            last_bucket_ids[tier] = current_id
+            continue
+        if current_id <= last_id:
+            continue
+
+        # Lazily decode the raw ring once across all tiers that fired.
+        if raw_rows is None:
+            try:
+                raw_items = redis_client.lrange(
+                    redis_key_for_tier("raw"), 0, GPS_TRENDS_RAW_MAX_SAMPLES - 1
+                ) or []
+            except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)